packages/react-instantsearch-hooks/src/lib/__tests__/useSearchState.test.tsx

```diff
@@ -0,0 +1,174 @@
+import { render } from '@testing-library/react';
+import { renderHook } from '@testing-library/react-hooks';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import {
+  InstantSearchHooksTestWrapper,
+  wait,
+  createInstantSearchTestWrapper,
+} from '../../../../../test/utils';
+import { useSearchBox } from '../../connectors/useSearchBox';
+import { useSearchState } from '../useSearchState';
+
+function SearchBox() {
+  const { query } = useSearchBox({});
+  return <>{query}</>;
+}
+
+describe('useSearchState', () => {
+  test('returns the ui state', async () => {
+    const wrapper = createInstantSearchTestWrapper();
+    const { result, waitForNextUpdate } = renderHook(() => useSearchState(), {
+      wrapper,
+    });
+
+    // Initial render state from manual `getWidgetRenderState`
+    expect(result.current).toEqual({
+      uiState: {
+        indexName: {},
+      },
+      indexUiState: {},
+      setUiState: expect.any(Function),
+      setIndexUiState: expect.any(Function),
+    });
+
+    const setUiState = result.current.setUiState;
+    const setIndexUiState = result.current.setIndexUiState;
+
+    await waitForNextUpdate();
+
+    // InstantSearch.js state from the `render` lifecycle step
+    expect(result.current).toEqual({
+      uiState: {
+        indexName: {},
+      },
+      indexUiState: {},
+      setUiState,
+      setIndexUiState,
+    });
+  });
+
+  test('returns the ui state with initial state', async () => {
+    const wrapper = createInstantSearchTestWrapper({
+      initialUiState: {
+        indexName: {
+          query: 'iphone',
+        },
+      },
+    });
+    const { result, waitForNextUpdate } = renderHook(() => useSearchState(), {
+      wrapper: ({ children }) =>
+        wrapper({
+          children: (
+            <>
+              <SearchBox />
+              {children}
+            </>
+          ),
+        }),
+    });
```

Comment:
The `<SearchBox>` is not needed in this test.

```suggestion
    const { result, waitForNextUpdate } = renderHook(() => useSearchState(), {
      wrapper: ({ children }) => wrapper({ children }),
    });
```